src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -76,6 +77,24 @@ object DrivesVehicle {
 
   case class StopDrivingIfNoPassengerOnBoardReply(success: Boolean, requestId: Int, tick: Int)
 
+  def processLinkEvents(eventsManager: EventsManager, vehicleId: Id[Vehicle], leg: BeamLeg): Unit = {
+    val path = leg.travelPath
+    if (path.linkTravelTime.nonEmpty & path.linkIds.size > 1) {
+      val links = path.linkIds
+      val linkTravelTime = path.linkTravelTime
+      var i: Int = 0
+      var curTime = leg.startTime
+      while (i < links.length - 1) {
```

Comment:
Should we keep the comment about why we are dropping the last item? Or methodize it in some way? Maybe that's where a test could come in handy?

Reply:
Let me bring the comment back. Here is some examples, hope it is helpful:

If you have `linkIds = Array(1, 2, 3)` and `linkTravelTime = Array(1, 2, 3)`, then the following events will be generated:
```
- LinkLeaveEvent at time 1.0, linkId = 1

- LinkEnterEvent at time 1.0, linkId = 2
- LinkLeaveEvent at time 3.0, linkId = 2

- LinkEnterEvent at time 3.0, linkId = 3
```
If you have `linkIds = Array(1, 2, 3, 4)` and `linkTravelTime = Array(1, 2, 3, 4)`:
```
- LinkLeaveEvent at time 1.0, linkId = 1

- LinkEnterEvent at time 1.0, linkId = 2
- LinkLeaveEvent at time 3.0, linkId = 2

- LinkEnterEvent at time 3.0, linkId = 3
- LinkLeaveEvent at time 6.0, linkId = 3

- LinkEnterEvent at time 6.0, linkId = 4
```
What I want to show is that we have to start with `LinkLeaveEvent` and stop with `LinkEnterEvent`. Because we stop with `LinkEnterEvent`, so the whole travel time through that link (last link) is dropped. Time is cumulative here. (It seems to me it is MATSim specific. @wrashid @colinsheppard? )